### ISSUE TITLE
Addon gets reset to default sometimes

### ISSUE THREAD

**Issue by skymen** [armandoalonso/c3IDE#63]
I got this a few times already when creating a new addon.
Steps are:
- Create new addon
- Work on it for a bit
- Restart server to upload it to C3 (Server was already running because I was working on another addon prior to that)
- Switch to another addon
- Switch back

This sometimes results in my work being erased and reset to the default new addon code. I usually just open C3's cached version and copy paste my code back and it fixes it most of the time, and I never get the issue with that addon again.

**Comment by armandoalonso** [comment]
that's interesting, starting the server from multiple addons should not impact the addon's code, the server is just a static file server pointing to C:\Users\{USER}\AppData\Roaming\C3IDE_DATA\Server\Test all addons get stored there by default when "compiling" the different parts of the addon. 

the way the app keeps state is a bit wonky sometimes (I need to figure out why and fix that) but after making modification do you save (ctrl-s), there is also an option to compile on save - which automatically generates the addon at the server path. as a work around while i figure that out, save when navigating to different tab.

**Comment by skymen** [comment]
Yeah, I already use compile on save, and I do save. And in fact, when I load the addon on C3 it loads what I worked on.
it's only when switching to another addon and switching back that I occasionally lose my stuff, even though it was indeed saved and hosted on the server
